File: DS/stack.cpp
//C program for linked list implementation of stack

#include<stdio.h>
#include<stdlib.h>
using namespace std;
struct Node
{
  int data;
  struct Node* next;
};

struct Node* newNode(int data)
{
  Node* node= new Node();
  node->data=data;
  node->next=NULL;
  return node;
}

int isEmpty(struct Node* root)
{
  return !root;
}

void push(struct Node** root, int data)
{
  Node* node=newNode(data);
  node->next=*root;
  *root=node;
  cout<<data;
}

int pop(struct Node** root)
{
  if (isEmpty(*root))
    return 0;
  Node* node=*root;
  *root=(*root)->next;
  int pop=node->data;
  free(node);
}

int peek(struct Node* root)
{
  if(isEmpty(*root))
    return 0;
  return root->data;
}

int main()
{
  Node* root=new Node();
  push(&root,10);
  push(&root,20);
  push(&root,30);
  printf("%d\n",pop(&root));
  printf("%d\n",peek(root));
  return 0;
}

File: random/python/Data-Structures/stack/stack.py
__author__ = "switchkiller"
class Stack:
    def __init__(self):
        self.content = []
    def push(self, item):
        self.content.append(item)
    def push(self):
        return self.content.pop()
    def isEmpty(self):
        return (self.items == [])
    def peek(self):
        return self.content
    def __str__(self):
        return str(self.content)
